### ISSUE TITLE
[BUG] Unhandled `OverflowException` from a Textbox Data bound to an Integer Property

### ISSUE THREAD

**Issue by WalkerCodeRanger** [reactiveui/ReactiveUI#3000]
**Describe the bug**
Databinding a textbox to an `int` causes an unhandled `OverflowException` when the user enters an out of range number (i.e. bigger than `int.MaxValue` or smaller than `int.MinValue`). This is inconsistent with the behavior when the user enters a non-integer value (e.g. "hello", or "3a4").

Note: I have a [StackOverflow question](https://stackoverflow.com/q/69816911/268898) related to this, but this part of the behavior appears to be a bug.

**Steps To Reproduce**

1. Have a property in your view model that is an `int`.
    ```csharp
    public class ExampleViewModel : ReactiveObject
    {
        [Reactive]
        public int Percentage { get; set; } = 10;
    }
    ```
2. Bind that property to the text of a textbox in your view.
    ```csharp
    public ExampleView : ReactiveControl<ExampleViewModel>
    {
        public ExampleView()
        {
            InitializeComponent();
            this.WhenActivated(d =>
            {
                this.Bind(ViewModel, vm => vm.Percentage, v => v.PercentageTextBox.Text).DisposeWith(d);
            });
        }
    }
    ```
3. Add an event handler to the application dispatcher unhandled exception event.
    ```xaml
    <Application x:Class="Example.App"
                 xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                 Startup="Application_Startup"
                 DispatcherUnhandledException="Application_DispatcherUnhandledException">
        <Application.Resources>
        </Application.Resources>
    </Application>
    ```
    ```csharp
    public partial class App : Application
    {
        private void Application_DispatcherUnhandledException(object sender, DispatcherUnhandledExceptionEventArgs e)
        {
            // TODO report the exception to the user somehow
        }
    }
    ```
4. Run the app and enter a value larger than `int.MaxValue` into the textbox.

When step 4 is done, the unhandled exception handler will be called with an `OverflowException` from the attempt to parse the input string into an `int`.

**Expected behavior**
No unhandled exception. The view model property is unchanged.

When a user enters a non-integer value (e.g. "3a4"), the `ComponentModelTypeConverter` catches the `FormatException` thrown from trying to parse the value into an `int`. It indicates the conversion failed instead of rethrowing the exception. There is no unhandled exception. The view model property is not changed. I would expect the same to happen in this case. 

**Environment**
- OS:       Windows 10 Pro
- Version: 10.0.19043 Build 19043
- Device:   Dell Laptop, Vostro 7590
- ReactiveUI Version:    13.0.38

**Additional context**
- .NET Framework 4.8
- WPF

**Comment by ChrisPulman** [comment]
Please try ReactiveUI Version 14.3.10 or newer

**Comment by glennawatson** [comment]
There were new converters added for integer values recently, so the version of RxUI you are using won't support it.

**Comment by WalkerCodeRanger** [comment]
I can confirm that it is fixed after we upgraded to 16.3.10.

Thanks for the quick response.
